kernel/arch/dreamcast/kernel/exec.c

```diff
@@ -48,7 +48,7 @@ void arch_exec_at(const void *image, uint32 length, uint32 address) {
         buffer[i] = _arch_exec_template[i];
 
     /* Plug in values */
-    values = buffer + (_arch_exec_template_values - _arch_exec_template);
+    values = buffer + ((uint32)_arch_exec_template_values - tstart);
```

Comment:
```suggestion
    values = buffer + ((uintptr_t)_arch_exec_template_values - tstart);
```
When casting a pointer to an integral type, the proper C type to use is `uintptr_t`, so that it's not architecture-specific. I know it is here, but I'm of the opinion it offers more clarity into the intention that the integer is an address.

Reply:
> When casting a pointer to an integral type, the proper C type to use is `uintptr_t`, so that it's not architecture-specific. I know it is here, but I'm of the opinion it offers more clarity into the intention that the integer is an address.

Yea, I did that elsewhere (for instance init.c) the reason for it here was that the file isn't currently using stdint types and is doing the same elsewhere for casting, so was just trying to solve that issue without cleaning up the whole thing. Will take it back and just clean up the whole thing, it's due.